2-mandatory/3-financial-times.js

```diff
@@ -4,36 +4,81 @@
     The home page of the web site has a headline section, which only has space for article titles which are 65 characters or less.
     Implement the function below, which will return a new array containing only article titles which will fit.
 */
-function potentialHeadlines(allArticleTitles) {
-    // TODO
+
+
+function potentialHeadlines(allArticleTitles){
+let shortTitles = [];
+let i = 0;
+
+while(i < allArticleTitles.length){
+    // console.log(ARTICLE_TITLES[i]);   
+    if (allArticleTitles[i].length < 66){
+        shortTitles.push(allArticleTitles[i]);
+        // console.log(shortTitles);
+    }    
+     i++;
+    //  console.log(shortTitles);     
+}return shortTitles;
 }
 
 /*
     The editor of the FT likes short headlines with only a few words!
     Implement the function below, which returns the title with the fewest words.
-    (you can assume words will always be seperated by a space)
-*/
+    (you can assume words will always be separated by a space)
+// */
+
 function titleWithFewestWords(allArticleTitles) {
-    // TODO
+  
+    let shortestTitle;
+    let shortestTitleLengthInWords;
+   
+    for (let i = 0; i < allArticleTitles.length; i++) {
+      const title = allArticleTitles[i];      
+      const titleLengthInWords = title.split(" ").length;  
+      if (
+        shortestTitle === undefined ||
+        shortestTitleLengthInWords > titleLengthInWords
+      ) {
+        shortestTitle = title;
+        shortestTitleLengthInWords = titleLengthInWords;
+      }
+    }  
+    return shortestTitle;
 }
 
+
+
 /*
     The editor of the FT has realised that headlines which have numbers in them get more clicks!
     Implement the function below to return a new array containing all the headlines which contain a number.
     (Hint: remember that you can also loop through the characters of a string if you need to)
 */
 function headlinesWithNumbers(allArticleTitles) {
-    // TODO
+    const titlesToReturn = [];
+  const numbers = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"];
+  allArticleTitles.forEach((title) => {
+    title.split("").forEach((character) => {
+      if (numbers.includes(character)) {
+        titlesToReturn.push(title);
+      }
+    });
+  });
+  return titlesToReturn;
 }
 
+
 /*
     The Financial Times wants to understand what the average number of characters in an article title is.
     Implement the function below to return this number - rounded to the nearest integer.
 */
-function averageNumberOfCharacters(allArticleTitles) {
-    // TODO
-}
 
+function averageNumberOfCharacters(allArticleTitles){
+    let sum = 0;
+    for (let i = 0; i < allArticleTitles.length; i++) {
+    sum += allArticleTitles[i];
```

Comment:
You've initialised `sum` to 0, but then you're essentially doing string concatenation of all the titles -- what were you actually trying to do here?

`allArticleTitles[i].length` would give you the number of characters in a given article.

Reply:
I'm not sure.